module-2/backend/week_one.md

```diff
@@ -5,30 +5,123 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
 11. What's responsive design?
+
+  - Responsive design refers to designing your styles (CSS and javascript) so that the page renders differently depending on the width of the browswer which usually varies between laptops, tablets, and mobile phones.
+
 12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
+
+  - Create
+  - Read
+  - Update
+  - Delete
+
+  - These actions are the four essential actions that you need to be able to execute on objects in your database. What if you could create things but you couldn't delete them? What if you could create things but you couldn't read them? Etc.
+
+13. What does HTTP stand for?
+
+  - Hypertext Transfer Protocol
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+
+  ```ruby
+  <%= @ruby_code %>
+  <% @ruby_code %>
+  ```
+  
+  - The first line prints the return value of the ruby code to the screen or inserts it into a string if it's surrounded by quotes. The second line just evaluates.
+
 15. What's an ORM?
+
+  - An Object Relational Mapper (ORM) is a relational database wrapper that helps the database and ruby classes talk to each other. Each table in the database represents a ruby class, each row in the table represents an instance of the class, and each column of the table represents an attribute of the class.
+
 16. What's the most commonly used ORM?
+
+  - ActiveRecord
+
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
+
+  - GET "/restaurants" --> See all restaurants
+  - GET "/restaurants/:id" --> See one restaurant
+  - GET "/restaurants/new" --> See form to create a restaurant
+  - POST "/restaurants" --> Submit to save new restaurant
+  - GET "/restaurants/:id/edit" --> See form to update restaurants
+  - PUT "/restaurants/:id" --> Submit to update restaurant
+  - DELETE "/restaurants/:id" --> Delete a restaurant
+
+18. What's a migration?
+
+  - A migration is a set of instructions to alter a database.
+
 19. When you create a migration, does it automatically modify your database?
+
+  - No, you have to actually run the migration for it to do so.
+
 20. How does a model relate to a database?
+
+  - A model sets the associations between tables in a database and sets validations for whether or not data gets stored to the database.

```

Comment:
One small clarification: using relationships like `has_many` and `belongs_to` doesn't actually set the associations in the database. The relationships at the DB level are set by adding foreign keys to your tables using migrations. Our models *rely on* those relationships to create methods for our models that allow us to use those as relationships between our model objects.